### PR TITLE
Shrink the size of ast::ConstantLit by 16 bytes.

### PR DESCRIPTION
**Pull request by jvilk-stripe**
Shrink the size of ast::ConstantLit by 16 bytes. Reduces Sorbet's max RSS on Stripe's codebase by 4.5%.

`resolutionScopes` is empty unless the constant failed to resolve. Then, `symbol` is set to `StubModule` and the scope at which the resolution failed is put into `resolutionScopes`.

We can remove another 8 bytes if we use a tagged pointer so that, when the symbol is resolved, we store the `symbol` in this field. Otherwise, we store a pointer to `resolutionScopes`.

Why 8 bytes? Although `symbol` is only 4 bytes, there is 4 bytes of padding between it and `resolutionScopes`!

### Motivation

Reduce memory consumption.

### Test plan

Covered by existing tests.

### CODE REVIEW

File: ast/Trees.cc [hunk 1]
```diff
@@ -686,10 +686,10 @@ string ConstantLit::showRaw(const core::GlobalState &gs, int tabs) {
     printTabs(buf, tabs + 1);
     fmt::format_to(std::back_inserter(buf), "orig = {}\n",
                    this->original ? this->original.showRaw(gs, tabs + 1) : "nullptr");
-    if (!resolutionScopes.empty()) {
+    if (resolutionScopes != nullptr && !resolutionScopes->empty()) {
```
Comment (jez) [inline ast/Trees.cc:689]: What do you think about ENFORCEing here that if `resolutionScopes` is not nullptr, that it's not empty?

Comment (jvilk-stripe) [reply ast/Trees.cc:689]: Works for me.

**Review by jez — APPROVED** [review]
lgtm, go nuts on adding ENFORCEs

File: ast/Trees.cc (hunk 1 above)
Comment (jvilk-stripe) [reply ast/Trees.cc:691]: Added. Also added to `_sanityCheck` on `ConstantLit`.

**Pull request merged**